### PR TITLE
Istio version from binary version

### PR DESCRIPTION
**Pull request by cebernardi**
**Please provide a description of this PR:**
This PR accomplishes #44937 by overriding `BootstrapNodeMetadata.IstioVersion` with the binary version if:
- ISTIO_META_ISTIO_VERSION environment variable is not set
- ISTIO_META_ISTIO_VERSION environment variable is set but empty

Istio Version environment variable setup was removed from:
- istio-start.sh
- Dockerfile.proxyv2

Also, golden files needed to be updated.

### CODE REVIEW

File: pkg/bootstrap/config.go [hunk 1]
```diff
@@ -560,6 +562,8 @@ func GetNodeMetaData(options MetadataOptions) (*model.Node, error) {
 		return nil, err
 	}
 
+	meta = SetIstioVersion(meta)
```
Comment (cebernardi) [inline pkg/bootstrap/config.go:565]: I'm not totally convinced by the approach of creating an additional function here...

I went down this route to separately test the logic, as `GetNodeMetaData` is a complex method and the related test covers a subset of the scenarios, and it seemed to me that adding only the additional IstioVersion scenarios to `TestGetNodeMetaData` made the test confusing, and trying to cover every scenarios was making the PR bigger and bigger. 

Thoughts?

File: pkg/bootstrap/config_test.go [hunk 2]
```diff
@@ -117,6 +119,48 @@ func TestGetNodeMetaData(t *testing.T) {
 	g.Expect(node.Metadata.Labels[model.LocalityLabel]).To(Equal("region/zone/subzone"))
 }
 
+func TestSetIstioVersion(t *testing.T) {
+	prevVersion := version.Info.Version
+	defer func() {
+		version.Info.Version = prevVersion
+	}()
```
Comment (howardjohn) [inline pkg/bootstrap/config_test.go:126]: nit: you can use `test.SetForTest`

**Review by howardjohn — APPROVED** [review]

**Comment by cebernardi** [comment]
Looking at the failures

File: pkg/bootstrap/config_test.go (hunk 2 above)
Comment (cebernardi) [reply pkg/bootstrap/config_test.go]: done

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by howardjohn** [comment]
I think we need to update TestProxyVersion in istio/release-builder to remove the check for the env var. Maybe we can replace it with a check doing `version` command and checking the output, but I would be fine with just opening an issue to add that as a followup as well

**Comment by cebernardi** [comment]
do you mean here? https://github.com/istio/release-builder/blob/master/pkg/validate/validate.go#L277

**Comment by howardjohn** [comment]
yep!

**Comment by cebernardi** [comment]
on it

**Review by howardjohn — COMMENTED** [review]
lgtm!

Thanks for the improvement. One last change needed:

```
prow/release-commit.sh
39:BUILDER_SHA=4712f3ce6f4edba2644af3e84292894af0055ded
```

Just needs to be updated to pull in your new release-builder change, then this should pass tests+merge

**Pull request merged**